### ISSUE TITLE
[Bug]: local flake install does not work

### ISSUE THREAD

**Issue by ragingpastry** [jetpack-io/devbox#956]
**Current Behavior (bug)**
When trying to use the php flake example in devbox I receive an error message

```
DEVBOX_DEBUG=1 devbox install                                                                                                                                  1 ↵
config.go:125: findProjectDir: path is 
config.go:166: finding devbox.json in dir: /home/crepe/projects/personal/devbox/examples/flakes/php
[DEBUG] 2023/04/26 07:58:04 go.jetpack.io/devbox/internal/boxcli/midcobra/debug.go:49: Debug mode enabled.
[DEBUG] 2023/04/26 07:58:04 go.jetpack.io/devbox/internal/boxcli/run.go:65: script: :
[DEBUG] 2023/04/26 07:58:04 go.jetpack.io/devbox/internal/boxcli/run.go:66: script args: []
[DEBUG] 2023/04/26 07:58:04 go.jetpack.io/devbox/internal/impl/config.go:125: findProjectDir: path is 
[DEBUG] 2023/04/26 07:58:04 go.jetpack.io/devbox/internal/impl/config.go:166: finding devbox.json in dir: /home/crepe/projects/personal/devbox/examples/flakes/php
Reinitialized existing Git repository in /home/crepe/projects/personal/devbox/examples/flakes/php/.devbox/gen/flake/.git/
Ensuring packages are installed.

Installing 2 packages: path:my-php-flake#php, path:my-php-flake#hello.

[1/2] path:my-php-flake#php
error: 'nixpkgs/f80ac848e3d6f0c12c52758c0f25c10c97ca3b62#path:my-php-flake#php' is not a valid URL
[1/2] path:my-php-flake#php: Fail
Error: Command: /run/current-system/sw/bin/nix profile install --profile /home/crepe/projects/personal/devbox/examples/flakes/php/.devbox/nix/profile/default --impure --priority 6 nixpkgs/f80ac848e3d6f0c12c52758c0f25c10c97ca3b62#path:my-php-flake#php --extra-experimental-features ca-derivations --option experimental-features nix-command flakes: exit status 1

[DEBUG] 2023/04/26 07:58:04 go.jetpack.io/devbox/internal/boxcli/midcobra/debug.go:70: Command stderr: 
[DEBUG] 2023/04/26 07:58:04 go.jetpack.io/devbox/internal/boxcli/midcobra/debug.go:72: 
ExecutionID:1f424c9e7d994e1683e25a60920965a3
exit status 1
Command: /run/current-system/sw/bin/nix profile install --profile /home/crepe/projects/personal/devbox/examples/flakes/php/.devbox/nix/profile/default --impure --priority 6 nixpkgs/f80ac848e3d6f0c12c52758c0f25c10c97ca3b62#path:my-php-flake#php --extra-experimental-features ca-derivations --option experimental-features nix-command flakes
go.jetpack.io/devbox/internal/nix.ProfileInstall
	go.jetpack.io/devbox/internal/nix/profile.go:223
go.jetpack.io/devbox/internal/impl.(*Devbox).addPackagesToProfile
	go.jetpack.io/devbox/internal/impl/packages.go:195
go.jetpack.io/devbox/internal/impl.(*Devbox).ensurePackagesAreInstalled
	go.jetpack.io/devbox/internal/impl/packages.go:140
go.jetpack.io/devbox/internal/impl.(*Devbox).RunScript
	go.jetpack.io/devbox/internal/impl/devbox.go:207
go.jetpack.io/devbox/internal/boxcli.runScriptCmd
	go.jetpack.io/devbox/internal/boxcli/run.go:74
go.jetpack.io/devbox/internal/boxcli.InstallCmd.func1
	go.jetpack.io/devbox/internal/boxcli/install.go:21
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.6.1/command.go:916
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.6.1/command.go:1044
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.6.1/command.go:968
go.jetpack.io/devbox/internal/boxcli/midcobra.(*midcobraExecutable).Execute
	go.jetpack.io/devbox/internal/boxcli/midcobra/midcobra.go:57
go.jetpack.io/devbox/internal/boxcli.Execute
	go.jetpack.io/devbox/internal/boxcli/root.go:78
go.jetpack.io/devbox/internal/boxcli.Main
	go.jetpack.io/devbox/internal/boxcli/root.go:91
main.main
	go.jetpack.io/devbox/cmd/devbox/main.go:11
runtime.main
	runtime/proc.go:250
runtime.goexit
	runtime/asm_amd64.s:1598
[DEBUG] 2023/04/26 07:58:04 go.jetpack.io/devbox/internal/impl/config.go:125: findProjectDir: path is 
[DEBUG] 2023/04/26 07:58:04 go.jetpack.io/devbox/internal/impl/config.go:166: finding devbox.json in dir: /home/crepe/projects/personal/devbox/examples/flakes/php
```

**Expected Behavior (fix)**
Expected the php example to work

**Additional context**
```
Version:     0.4.7
Platform:    linux_amd64
Commit:      none
Commit Time: unknown
Go Version:  go1.20.3
```

**Comment by ragingpastry** [comment]
I have a feeling this is a problem with either NixOS or my local environment somehow... as this works fine on devbox.sh cloud. Trying to investigate further

**Comment by ragingpastry** [comment]
Works just fine in a `nixos/nix` docker container! The issue must be in my local environment. Apologies for the spam, closing ticket while I debug my local environment

**Comment by ragingpastry** [comment]
For completeness I think I figured out what the issue was, however I am still not entirely sure why it is happening.

I have a custom nixpkg for devbox because that latest devbox version is not released in any nixos channel yet. In this Nixpkg, I simply copied the one upstream: https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/tools/devbox/default.nix#L38 and modified the version number. I did NOT change the hash. After building the package I was able to verify that `devbox version -v` returned with 0.4.8. However, I think that I was actually on version 0.4.6 still.

Perhaps NixOS was caching the resulting derivation because of the hash being the same, and simply supplying the updated 0.4.8 version string to the package build? Not entirely sure about this.

In order to resolve, I modified the hash to an invalid hash, got nix to spit out the correct hash, and then put the correct hash into the pkg derivation. Everything works as expected now!
